src/main/java/uk/gov/ons/census/action/schedule/ActionInstructionBuilder.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.ons.census.action.schedule;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.action.model.dto.instruction.ActionAddress;
+import uk.gov.ons.census.action.model.dto.instruction.ActionEvent;
+import uk.gov.ons.census.action.model.dto.instruction.ActionInstruction;
+import uk.gov.ons.census.action.model.dto.instruction.ActionRequest;
+import uk.gov.ons.census.action.model.dto.instruction.Priority;
+import uk.gov.ons.census.action.model.entity.ActionRule;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.entity.UacQidLink;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+@Component
+public class ActionInstructionBuilder {
+  private final UacQidLinkRepository uacQidLinkRepository;
+
+  @Value("${queueconfig.outbound-exchange}")
+  private String outboundExchange;
+
+  public ActionInstructionBuilder(UacQidLinkRepository uacQidLinkRepository) {
+    this.uacQidLinkRepository = uacQidLinkRepository;
+  }
+
+  public ActionInstruction buildActionInstruction(Case caze, ActionRule actionRule) {
+
+    List<UacQidLink> uacQidLinks = uacQidLinkRepository.findByCaseId(caze.getCaseId().toString());
+
+    UacQidLink uacQidLink;
+    Optional<UacQidLink> uacQidLinkWales = Optional.ofNullable(null);
+
+    if (uacQidLinks == null || uacQidLinks.isEmpty()) {
+      throw new RuntimeException(); // TODO: How can we process this case without a UAC?
+    } else if (uacQidLinks.size() > 1) {
+      if (isQuestionnaireWelsh(caze.getTreatmentCode()) && uacQidLinks.size() == 2) {
+        uacQidLink = getSpecificUacQidLinkByQuestionnaireType(uacQidLinks, "02", "03");
```

Comment:
These "02", "03" would be better as Constants to make clear what they are, especially as they're re-used